src/controllers/controller.h

```diff
@@ -19,13 +19,15 @@
 #include "controllers/controllervisitor.h"
 #include "controllers/engine/controllerengine.h"
 #include "util/duration.h"
+#include "util/parented_ptr.h"
 
+class ControlObject;
 class ControllerJSProxy;
 
 class Controller : public QObject, ConstControllerPresetVisitor {
     Q_OBJECT
   public:
-    Controller();
+    Controller(const QString& group);
```

Comment:
explicit

Minor improvement: Since Controller is supposed to be an abstract base class we could reduce the visibility of the ctor from public to protected.